tangram.ts

```diff
@@ -0,0 +1,88 @@
+import bun from "bun" with { path: "../packages/packages/bun" };
+import { cargo } from "rust" with { path: "../packages/packages/rust" };
+import * as std from "std" with { path: "../packages/packages/std" };
+import { $ } from "std" with { path: "../packages/packages/std" };
+
+import cargoToml from "./Cargo.toml" with { type: "file" };
+import cargoLock from "./Cargo.lock" with { type: "file" };
+import biomeJson from "./biome.json" with { type: "file" };
+import bunLockb from "./bun.lockb" with { type: "file" };
+import packageJson from "./package.json" with { type: "file" };
+import packages from "./packages" with { type: "directory" };
+
+export const source = tg.target(() =>
+	tg.directory({
+		"Cargo.toml": cargoToml,
+		"Cargo.lock": cargoLock,
+		"biome.json": biomeJson,
+		"bun.lockb": bunLockb,
+		"package.json": packageJson,
+		packages: packages,
+	}),
+);
+
+export default tg.target(async () => {
+	const host = std.triple.host();
+	const env = std.env.arg(
+		bun({ host }),
+		librustyv8(host),
+		linuxRuntimeComponents(),
+	);
+	return cargo.build({
+		buildInTree: true,
+		checksum: "unsafe",
+		source: source(),
+		env,
+		parallelJobs: 4,
+	});
+});
+
+export const librustyv8 = tg.target(async (hostArg?: string) => {
+	const host = hostArg ?? (await std.triple.host());
+	let os;
+	if (std.triple.os(host) === "darwin") {
+		os = "apple-darwin";
+	} else if (std.triple.os(host) === "linux") {
+		os = "unknown-linux-gnu";
+	} else {
+		throw new Error(`unsupported host ${host}`);
+	}
+	const checksum = "unsafe";
+	const file = `librusty_v8_release_${std.triple.arch(host)}-${os}.a.gz`;
+	const lib = await std
+		.download({
+			checksum,
+			decompress: true,
+			url: `https://github.com/denoland/rusty_v8/releases/download/v129.0.0/${file}`,
```

Comment:
Please pull out the version to a const.

Reply:
added logic to instead read the version of the `v8` crate in `cargo.Lock`